hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/grantapproval/ClassicGrantApprovalCall.java

```diff
@@ -70,7 +76,35 @@ public PricedResult execute() {
                     ? GrantApprovalTranslator.GRANT_APPROVAL.getOutputs().encodeElements((long) status.protoOrdinal())
                     : GrantApprovalTranslator.GRANT_APPROVAL_NFT.getOutputs().encodeElements((long)
                             status.protoOrdinal());
+
+            final var tokenAddress = asLongZeroAddress(token.tokenNum());
+            if (tokenType.equals(TokenType.FUNGIBLE_COMMON)) {
+                frame.addLog(getLogForFungibleAdjustAllowance(tokenAddress));
+            } else {
+                frame.addLog(getLogForNftAdjustAllowance(tokenAddress));
+            }
+
             return gasOnly(FullResult.successResult(encodedOutput, gasRequirement));
         }
     }
+
+    private Log getLogForFungibleAdjustAllowance(final Address logger) {
+        return EncodingFacade.LogBuilder.logBuilder()
```

Comment:
⚠️ We don't want to leave any dependency on the `mono-service` module floating around; so we need to re-create this utility somewhere in the `contract-service-impl` module.